linkerd/app/inbound/src/endpoint.rs

```diff
@@ -37,12 +43,44 @@ pub struct TcpEndpoint {
 
 #[derive(Clone, Debug)]
 pub struct RequestTarget {
-    accept: tls::accept::Meta,
+    accept: TcpAccept,
 }
 
 #[derive(Copy, Clone, Debug)]
 pub struct ProfileTarget;
 
+// === impl TcpAccept ===
+
+impl From<listen::Addrs> for TcpAccept {
+    fn from(tcp: listen::Addrs) -> Self {
+        Self {
+            target_addr: tcp.target_addr(),
+            peer_id: tls::Conditional::None(tls::ReasonForNoPeerName::PortSkipped.into()),
+        }
+    }
+}
+
+impl From<tls::accept::Meta> for TcpAccept {
+    fn from(tls: tls::accept::Meta) -> Self {
+        Self {
+            target_addr: tls.addrs.target_addr(),
+            peer_id: tls.peer_identity,
+        }
+    }
+}
+
+impl Into<SocketAddr> for &'_ TcpAccept {
+    fn into(self) -> SocketAddr {
+        self.target_addr
+    }
+}
+
+impl Into<transport::labels::Key> for &'_ TcpAccept {
+    fn into(self) -> transport::labels::Key {
+        transport::labels::Key::accept(transport::labels::Direction::In, self.peer_id.clone())
+    }
+}
```

Comment:
elsewhere i've noticed comments explaining where the various `From`/`Into` impls get used...it might be nice to have those here?

Reply:
...although, i suppose that when it's labels, it's pretty obvious what it's for...